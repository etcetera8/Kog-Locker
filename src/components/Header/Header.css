.header-container {
  text-align: center;
  border-bottom: 2px solid #fff;
  height: 100px;
  /*  background-image: url('../../assets/topo.png');
*/ }
  .header-container .welcome {
    color: #fff;
    margin-top: 40px;
    margin-right: 5px;
    float: right; }
  .header-container .profile-img {
    border: 2px solid #fff;
    margin-right: 20px;
    border-radius: 50%;
    float: right;
    clear: both; }
  .header-container .page-title {
    color: #fff;
    font-size: 2.2em;
    letter-spacing: 2px;
    background: #ff6400;
    /*For browsers that don't support gradients*/
    background: linear-gradient(to right, transparent, #ff6400);
    width: 90%;
    margin: 35px auto 0 auto;
    padding: 11.5px;
    border-top-right-radius: 20px;
    border-bottom-right-radius: 20px; }
  .header-container .cog {
    color: #ff6400; }
