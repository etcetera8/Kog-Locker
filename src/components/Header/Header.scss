

.header-container {
  text-align: center;
  border-bottom: 1px solid #000;
  height: 100px;

  .welcome {
    margin-right: 15px;
    float: right;
    clear: both;
  }

  .profile-img {
    margin-right: 15px;
    border-radius: 50%;
    float: right;
    clear: both;
  }

}