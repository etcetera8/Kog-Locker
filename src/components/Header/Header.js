import React from 'react';
import './Header.css';
import PropTypes from 'prop-types';

const Header = ({userName, userPicture}) => {
  return (
    <header className="header-container"> 
      <img className="profile-img" src={userPicture} alt="USER" />
      <span className="welcome">Welcome, {userName}</span>
      <h1 className="page-title">C<span className="cog">o</span>g L<span className="cog">o</span>ckr</h1> 
    </header>
  );
};

Header.propTypes = {
  userName: PropTypes.string,
  userPicture: PropTypes.string
};

export default Header;