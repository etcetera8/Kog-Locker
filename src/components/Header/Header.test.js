import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

describe('header', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot();
  })

})