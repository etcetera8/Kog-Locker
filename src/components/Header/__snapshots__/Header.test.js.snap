// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="profile picture"
          className="profile-img"
          src={undefined}
/>,
        <span
          className="welcome"
>
          Welcome, 
</span>,
        <h1
          className="page-title"
>
          C
          <span
                    className="cog"
          >
                    o
          </span>
          g L
          <span
                    className="cog"
          >
                    o
          </span>
          ckr
</h1>,
      ],
      "className": "header-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "profile picture",
          "className": "profile-img",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome, ",
            undefined,
          ],
          "className": "welcome",
        },
        "ref": null,
        "rendered": Array [
          "Welcome, ",
          undefined,
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "C",
            <span
              className="cog"
>
              o
</span>,
            "g L",
            <span
              className="cog"
>
              o
</span>,
            "ckr",
          ],
          "className": "page-title",
        },
        "ref": null,
        "rendered": Array [
          "C",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "o",
              "className": "cog",
            },
            "ref": null,
            "rendered": "o",
            "type": "span",
          },
          "g L",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "o",
              "className": "cog",
            },
            "ref": null,
            "rendered": "o",
            "type": "span",
          },
          "ckr",
        ],
        "type": "h1",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="profile picture"
            className="profile-img"
            src={undefined}
/>,
          <span
            className="welcome"
>
            Welcome, 
</span>,
          <h1
            className="page-title"
>
            C
            <span
                        className="cog"
            >
                        o
            </span>
            g L
            <span
                        className="cog"
            >
                        o
            </span>
            ckr
</h1>,
        ],
        "className": "header-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "profile picture",
            "className": "profile-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome, ",
              undefined,
            ],
            "className": "welcome",
          },
          "ref": null,
          "rendered": Array [
            "Welcome, ",
            undefined,
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "C",
              <span
                className="cog"
>
                o
</span>,
              "g L",
              <span
                className="cog"
>
                o
</span>,
              "ckr",
            ],
            "className": "page-title",
          },
          "ref": null,
          "rendered": Array [
            "C",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "o",
                "className": "cog",
              },
              "ref": null,
              "rendered": "o",
              "type": "span",
            },
            "g L",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "o",
                "className": "cog",
              },
              "ref": null,
              "rendered": "o",
              "type": "span",
            },
            "ckr",
          ],
          "type": "h1",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
